### ISSUE TITLE
odoo.addons.hw_escpos.controllers.main: write() got multiple values for keyword argument 'timeout'

### ISSUE THREAD

**Issue by ata2001** [odoo/odoo#19017]
Impacted versions:

10.0

Steps to reproduce:

Connect usb printer, enable hw_escpos.

Current behavior:

Odoo fails with error.

Log:

```
2017-08-23 14:37:18,903 11242 INFO ? odoo.addons.hw_escpos.controllers.main: connected : Connected to Љ Љ (in=0x82,out=0x01)
2017-08-23 14:37:18,927 11242 INFO ? odoo.addons.hw_escpos.controllers.main: error : write() got multiple values for keyword argument 'timeou
t'
2017-08-23 14:37:18,927 11242 ERROR ? odoo.addons.hw_escpos.controllers.main: ESC/POS Error: write() got multiple values for keyword argument
 'timeout'
2017-08-23 14:37:18,927 11242 ERROR ? odoo.addons.hw_escpos.controllers.main: write() got multiple values for keyword argument 'timeout'
------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ata/Softwares/odoo/addons/hw_escpos/controllers/main.py", line 167, in run
    self.print_status(printer)
  File "/home/ata/Softwares/odoo/addons/hw_escpos/controllers/main.py", line 201, in print_status
    eprint.text('\n\n')
  File "/home/ata/Softwares/odoo/addons/hw_escpos/escpos/escpos.py", line 824, in text
    self._raw(txt)
  File "/home/ata/Softwares/odoo/addons/hw_escpos/escpos/printer.py", line 85, in _raw
    if len(msg) != self.device.write(self.out_ep, msg, self.interface, timeout=5000):
TypeError: write() got multiple values for keyword argument 'timeout'
------------------------------------------------------------
```

Expected behavior:

Odoo successfully connects to the printer and prints status message.

---------------------------------------------------------------------------------------------------------

The pyusb API has changed in 1.0.0b2. `write()` dropped the `interface` argument.

A workaround was to change the version number of pyusb to 1.0.0b1 in the requirements.txt. (currently it is 1.0.0)

**Comment by Yenthe666** [comment]
Hi @ata2001,

Thank you for your report and the detailed information.
@jorenvo perhaps we have to change this?

Regards,
Yenthe

**Comment by lem8r** [comment]
Removing timeout parameter here https://github.com/odoo/odoo/blob/master/addons/hw_escpos/escpos/printer.py#L86
fixes this issue.

**Comment by ata2001** [comment]
@lem8r It may have worked for you, but that's not the correct solution. The `interface` argument was dropped, if you remove timeout, `self.interface` is going to get parsed as timeout.

**Comment by pedrobaeza** [comment]
Is this still happening?

**Comment by rachmataditiya** [comment]
yes, still happening on odoo 12 also @pedrobaeza

**Comment by azmeuk** [comment]
I can confirm it happens with :

- odoo 8 (posbox)
- Ubuntu 19.04
- Python 2.7.16
- pyusb 1.0.2
- pyserial 3.4

**Comment by azmeuk** [comment]
For the record, [here is](https://github.com/pyusb/pyusb/commit/20cd8c1f79b24082ec999c022b56c3febedc0964#diff-b5a4f98a864952f0f55d569dd14695b7L293) the commit where the 'interface' parameter has been deleted.

**Comment by nick2474** [comment]
> Removing timeout parameter here https://github.com/odoo/odoo/blob/master/addons/hw_escpos/escpos/printer.py#L86
> fixes this issue.

**Scenario**:
Raspberry PI 4, Raspbian GNU/Linux 10 (buster) 
I spent almos 4 hours trying to solve the same issue. Thanks to this post i fixed the issue.
Now i got its working with this script previous running https://gitlab.com/-/snippets/1723092.
Thanks so much!

**Comment by azmeuk** [comment]
This ticket is fixed by #38390

**Comment by pedrobaeza** [comment]
Thanks for the follow-up, so closing.
